x/nexus/keeper/lockable_asset.go

```diff
@@ -106,36 +93,42 @@ func (c lockableAsset) UnlockTo(ctx sdk.Context, toAddr sdk.AccAddress) error {
 func (c lockableAsset) getCoin(ctx sdk.Context) (sdk.Coin, error) {
 	switch c.coinType {
 	case types.ICS20:
-		return c.toICS20(ctx)
+		return toICS20(ctx, c.nexus, c.ibc, c.Coin)
```

Comment:
why remove the associate method?

Reply:
Because not all coin types could be converted to ICS20. And so we ended up having to recheck the coin type in the method. Instead `toICS20` is a separate helper method that should only be called when needed.
```
	if c.coinType != types.ICS20 {
		return sdk.Coin{}, fmt.Errorf("%s is not ICS20 token", c.GetDenom())
	}
```